test/test_genty.py

```diff
@@ -493,3 +493,17 @@ class SomeClass(object):
         instance = SomeClass()
 
         self.assertIn('test_defined_in_metaclass({0})'.format(repr('foo')), dir(instance))
+
+    def test_dataprovider_returning_genty_args_passes_correct_args(self):
+        @genty
+        class TestClass(object):
+            def builder(self):
+                return genty_args(42, named='named_arg')
+
+            @genty_dataprovider(builder)
+            def test_method(self, number, default=None, named=None):
+                return number, default, named
+
+        instance = TestClass()
+        # pylint:disable=no-member
+        self.assertItemsEqual(instance.test_method_builder(), (42, None, 'named_arg'))
```

Comment:
I believe that in most asserts we put the 'expected' data first, followed by the actual data. So I'd swap the order of the 2 params.